common/rpc/grpc.go

```diff
@@ -123,10 +121,7 @@ func Dial(hostName string, tlsConfig *tls.Config, logger log.Logger, interceptor
 		grpc.WithConnectParams(cp),
 	}
 
-	return grpc.Dial(
-		hostName,
-		dialOptions...,
-	)
+	return grpc.NewClient(hostName, dialOptions...)
```

Comment:
Also, according to the docs there are some subtle differences between Dial/DialContext and NewClient (the default resolver thing). Do we think our existing testing is enough to catch any issues from that?

Reply:
replace DialContext except for otel

Reply:
I think the cluster test covers this, not unit tests.